src/main/java/com/google/flourbot/api/Bot.java

```diff
@@ -57,26 +64,75 @@ public static void main(String[] args) {
   @PostMapping("/")
   public void onEvent(@RequestBody JsonNode event) throws IOException, GeneralSecurityException {
     switch (event.at("/type").asText()) {
+
       case "ADDED_TO_SPACE":
         String spaceType = event.at("/space/type").asText();
         if ("ROOM".equals(spaceType)) {
           String displayName = event.at("/space/displayName").asText();
           replyText = String.format("Thanks for adding me to %s. Type \"@MacroBot help\" at any time to see my instructions.", displayName);
+          helpMessage = "To use a room member's macro, the creator of a macro must write \"@MacroBot initiate MacroName \". " + 
+            "Once this has been sent, any room member can use the initialized macro by writing \"@MacroBot MacroName <your message> \" " + 
+            "If your macro has already been used in a thread, you may omit the MacroName and can simply write \"@MacroBot <your message>\".";
+
         } else {
           String displayName = event.at("/user/displayName").asText();
-          replyText = String.format("Thanks for adding me to a DM, %s!", displayName);
+          replyText = String.format("Thanks for adding me to a DM, %s! Type \"@MacroBot help\" at any time to see my instructions.", displayName);
+          helpMessage = "To use one of your macros, simply write \"@MacroBot MacroName <your message> \".";
         }
         break;
+
       case "MESSAGE":
-        // Sends request to execution module
-        String email = event.at("/message/sender/email").asText();
         String message = event.at("/message/text").asText();
+        String[] words = message.split(" ");
         String threadId = event.at("/message/thread/name").asText();
-        replyText = macroExecutionModule.execute(email, message, threadId);
+
+        // Check that a message was written.
+        if (words.length <= 1) {
+            replyText = "You must type a message when you message me. Please type \"@MacroBot help\" for more instructions.";
+            break;
+        }
+        // Reply to the help message.
+        if (words[1].equalsIgnoreCase("help")) {  
+            replyText = helpMessage; 
+        }        
+        else if (words.length >= 2) {
+            // If a macro is being initialized for a room.
+            if (words[1].equalsIgnoreCase("initiate")) {
+                // getMacroName() requires the macroName to be the second word of the message, so remove the word "initiate". This will put the macro name as the second word in the message.
+                List<String> wordsWithoutInitiate = new LinkedList(Arrays.asList(words));
+                wordsWithoutInitiate.remove("initiate");
```

Comment:
If word 1 is indeed "initiate",  taking and joining the sub array from index [2: end] might be easier.

Reply:
Could you please elaborate on how this is done? I am looking to remove the first element of the String array but keep everything else.

Reply:
I missed the fact that we still need element 0